File: myutils-orm/src/main/java/org/easyarch/myutils/test/dao/UserMapper.java
package org.easyarch.myutils.test.dao;

import org.easyarch.myutils.orm.annotation.sql.SqlParam;
import org.easyarch.myutils.test.pojo.User;

import java.util.List;

/**
 * Description :
 * Created by xingtianyu on 17-2-9
 * 上午2:07
 * description:
 */

public interface UserMapper {

    public User findById(@SqlParam(name = "id") String id);

    public List<User> findByUser(User user);
}

File: myutils-orm/src/main/resources/mapper/usermapper.js
/**
 * Description :
 * Created by code4j on 17-1-28
 *  上午1:35
 */
ctx.namespace = "org.easyarch.myutils.test.dao.UserMapper";

function findById(params){
    return "select * from user where client_id = $id$";
}

function findByUser(params) {
    var sql = "select * from user" + ctx.where;
    if (params.clientId != undefined){
        sql += "client_id = $clientId$ and";
    }
    if (params.phone != undefined){
        sql += "phone = phone";
    }
    return sql;
}